test/test.redispatch.js

```diff
@@ -0,0 +1,89 @@
+import QUnit from 'qunit';
+import videojs from 'video.js';
+import redispatch from '../src/redispatch.js';
+
+QUnit.module('Redispatch', {
+
+  beforeEach(assert) {
+
+    // Player event buffer.
+    // Mocked player pushes events here when they are triggered.
+    // redispatch helper returns event buffer after each redispatch.
+    let eventBuffer = [];
+
+    // Mocked player
+    this.player = {
+      trigger(event) {
+        eventBuffer.push(event);
+      },
+
+      currentSrc() {
+        return 'my vid';
+      },
+
+      ads: {
+        state: 'content-set',
+
+        snapshot: {
+          ended: false,
+          currentSrc: 'my vid'
+        },
+
+        videoElementRecycled() {
+          return false;
+        },
+
+        stitchedAds() {
+          return false;
+        }
+      }
+    };
+
+    // Redispatch helper for tests
+    this.redispatch = function(type) {
+      const event = {type};
+
+      eventBuffer = [];
+      redispatch.call(this.player, event);
+
+      if (eventBuffer.length === 1) {
+        return eventBuffer[0].type;
+      } else if (event.cancelBubble) {
```

Comment:
Redispatch can add cancelBubble to the mocked event, even though it's not a real event.

I added a new unit test to this PR to show that it works.